### PR TITLE
update_CRUD_Plate

### PR DESCRIPTION
**Pull request by BelisAliosmanova**
Changes made:
- change the model mapper
- handle exceptions in one global handler
- add more checks in update and create method
- make safe delete
-create more tests covering 100% of the crud

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,22 @@
 		<spring-cloud.version>2022.0.3</spring-cloud.version>
 	</properties>
 	<dependencies>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<version>5.3.1</version>
+		</dependency>
+		<dependency>
```
Comment (emiliyank) [inline pom.xml:26]: jacoco is already added and is in the main branch

File: pom.xml [hunk 2]
```diff
@@ -83,6 +99,25 @@
 
 	<build>
 		<plugins>
+			<plugin>
```
Comment (emiliyank) [inline pom.xml:102]: already added in main branch

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/exeptions/ApiPackageNotFound.java [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 import java.time.ZonedDateTime;
 
+//TODO да се обедини apiException за всички класове !
```
Comment (emiliyank) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/exeptions/ApiPackageNotFound.java:7]: remove this comment, and probably remove the whole ApiPackageNotFound

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java [hunk 4]
```diff
@@ -1,144 +1,270 @@
...
+        verifyNoInteractions(modelMapper);
     }
 
     @Test
-    void testGetCartonById_NonExistingCarton() {
-        when(cartonRepository.findById(any(Long.class))).thenReturn(Optional.empty());
-        assertThrows(ApiRequestException.class, () -> {
-            cartonService.getCartonById(1L);
-        });
+    public void testCreateCarton_InvalidCartonDTO_NameMissing() {
```
Comment (emiliyank) [inline src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java:105]: Do we have a test method that checks for a successful creation of Carton?

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java [hunk 5]
```diff
@@ -1,144 +1,270 @@
...
+        existingCarton.setSize("Existing Carton");
+        existingCarton.setAvailableQuantity(5);
+        existingCarton.setPrice(50.0);
+
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.of(existingCarton));
+        assertThrows(ValidationException.class, () -> cartonService.updateCarton(1L, cartonDTO));
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verifyNoInteractions(modelMapper);
     }
+    @Test
```
Comment (emiliyank) [inline src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java:233]: is there a test method that check for successful update of a carton?

**Review by emiliyank — COMMENTED** [review]
In general looks very good. There are lots of good improvements done.
Usually for the migrations we would create a new migration that adds the is_deleted column. But in this case since the project is in very early stage and is not deployed on server we can leave as it is now.

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java (hunk 4 above)
Comment (BelisAliosmanova) [reply src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java:105]: Yes, we have that test but in other branch which is also in pull-request

File: src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java (hunk 5 above)
Comment (BelisAliosmanova) [reply src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java:233]: Yes, we have that test but in other branch which is also in pull-request

**Review by emiliyank — APPROVED** [review]

**Review by berki0 — COMMENTED** [review]
every thing looks good !

**Pull request merged**